## Project - Seattle Airbnb Dataset  

BY Fabi Daniel

In [1]:
# import all necessary modules
import numpy as np
print('numpy: {}'.format(np.__version__))
import pandas as pd
print('pandas: {}'.format(pd.__version__))
from matplotlib import pyplot as plt
# print('matplotlib: {}'.format(matplotlib.__version__))
import seaborn as sns
print('seaborn: {}'.format(sns.__version__))
%matplotlib inline

numpy: 1.23.1
pandas: 1.4.3
seaborn: 0.11.2


In [2]:
#load dataset

calendar_df = pd.read_csv('/Users/cwrealestate/Downloads/archive 2/calendar.csv')
listings_df = pd.read_csv('/Users/cwrealestate/Downloads/archive 2/listings.csv')
reviews_df = pd.read_csv('/Users/cwrealestate/Downloads/archive 2/reviews.csv')

In [3]:
# Get overview of Calendar
calendar_df.head()

listing_id        date available   price
0      241032  2016-01-04         t  $85.00
1      241032  2016-01-05         t  $85.00
2      241032  2016-01-06         f     NaN
3      241032  2016-01-07         f     NaN
4      241032  2016-01-08         f     NaN

In [4]:
# Get info 
calendar_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1393570 entries, 0 to 1393569
Data columns (total 4 columns):
 #   Column      Non-Null Count    Dtype 
---  ------      --------------    ----- 
 0   listing_id  1393570 non-null  int64 
 1   date        1393570 non-null  object
 2   available   1393570 non-null  object
 3   price       934542 non-null   object
dtypes: int64(1), object(3)
memory usage: 42.5+ MB


* There are **4** Columns In calendar_df, **1,393,570** observations
* ['price'] with missing values

In [5]:
# See listings dataframe
listings_df.head()

id                           listing_url       scrape_id last_scraped  \
0   241032   https://www.airbnb.com/rooms/241032  20160104002432   2016-01-04   
1   953595   https://www.airbnb.com/rooms/953595  20160104002432   2016-01-04   
2  3308979  https://www.airbnb.com/rooms/3308979  20160104002432   2016-01-04   
3  7421966  https://www.airbnb.com/rooms/7421966  20160104002432   2016-01-04   
4   278830   https://www.airbnb.com/rooms/278830  20160104002432   2016-01-04   

                                  name  \
0         Stylish Queen Anne Apartment   
1   Bright & Airy Queen Anne Apartment   
2  New Modern House-Amazing water view   
3                   Queen Anne Chateau   
4       Charming craftsman 3 bdm house   

                                             summary  \
0                                                NaN   
1  Chemically sensitive? We've removed the irrita...   
2  New modern house built in 2013.  Spectacular s...   
3  A charming apartment that sits atop Queen Anne...   
4  Cozy family craftman house in beautiful neighb...   

                                               space  \
0  Make your self at home in this charming one-be...   
1  Beautiful, hypoallergenic apartment in an extr...   
2  Our house is modern, light and fresh with a wa...   
3                                                NaN   
4  Cozy family craftman house in beautiful neighb...   

                                         description experiences_offered  \
0  Make your self at home in this charming one-be...                none   
1  Chemically sensitive? We've removed the irrita...                none   
2  New modern house built in 2013.  Spectacular s...                none   
3  A charming apartment that sits atop Queen Anne...                none   
4  Cozy family craftman house in beautiful neighb...                none   

                               neighborhood_overview  ... review_scores_value  \
0                                                NaN  ...                10.0   
1  Queen Anne is a wonderful, truly functional vi...  ...                10.0   
2  Upper Queen Anne is a charming neighborhood fu...  ...                10.0   
3                                                NaN  ...                 NaN   
4  We are in the beautiful neighborhood of Queen ...  ...                 9.0   

  requires_license license jurisdiction_names instant_bookable  \
0                f     NaN         WASHINGTON                f   
1                f     NaN         WASHINGTON                f   
2                f     NaN         WASHINGTON                f   
3                f     NaN         WASHINGTON                f   
4                f     NaN         WASHINGTON                f   

  cancellation_policy  require_guest_profile_picture  \
0            moderate                              f   
1              strict                              t   
2              strict                              f   
3            flexible                              f   
4              strict                              f   

  require_guest_phone_verification calculated_host_listings_count  \
0                                f                              2   
1                                t                              6   
2                                f                              2   
3                                f                              1   
4                                f                              1   

  reviews_per_month  
0              4.07  
1              1.48  
2              1.15  
3               NaN  
4              0.89  

[5 rows x 92 columns]

In [6]:
# Get info
listings_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3818 entries, 0 to 3817
Data columns (total 92 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   id                                3818 non-null   int64  
 1   listing_url                       3818 non-null   object 
 2   scrape_id                         3818 non-null   int64  
 3   last_scraped                      3818 non-null   object 
 4   name                              3818 non-null   object 
 5   summary                           3641 non-null   object 
 6   space                             3249 non-null   object 
 7   description                       3818 non-null   object 
 8   experiences_offered               3818 non-null   object 
 9   neighborhood_overview             2786 non-null   object 
 10  notes                             2212 non-null   object 
 11  transit                           2884 non-null   object 
 12  thumbn

* There are **91** columns in listings_df, **3,818** observations

In [7]:
# See reviews
reviews_df.head()

listing_id        id        date  reviewer_id reviewer_name  \
0     7202016  38917982  2015-07-19     28943674        Bianca   
1     7202016  39087409  2015-07-20     32440555         Frank   
2     7202016  39820030  2015-07-26     37722850           Ian   
3     7202016  40813543  2015-08-02     33671805        George   
4     7202016  41986501  2015-08-10     34959538          Ming   

                                            comments  
0  Cute and cozy place. Perfect location to every...  
1  Kelly has a great room in a very central locat...  
2  Very spacious apartment, and in a great neighb...  
3  Close to Seattle Center and all it has to offe...  
4  Kelly was a great host and very accommodating ...

In [9]:
# Get Info
reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 84849 entries, 0 to 84848
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   listing_id     84849 non-null  int64 
 1   id             84849 non-null  int64 
 2   date           84849 non-null  object
 3   reviewer_id    84849 non-null  int64 
 4   reviewer_name  84849 non-null  object
 5   comments       84831 non-null  object
dtypes: int64(3), object(3)
memory usage: 3.9+ MB


* There are **6** columns in reviews_df, **84,849** observations

### Data Assessment


In [10]:
listings_df.sample(10)

id                            listing_url       scrape_id  \
1528  10125020  https://www.airbnb.com/rooms/10125020  20160104002432   
2012   4693488   https://www.airbnb.com/rooms/4693488  20160104002432   
1067    938785    https://www.airbnb.com/rooms/938785  20160104002432   
2556   3646393   https://www.airbnb.com/rooms/3646393  20160104002432   
2767   1686958   https://www.airbnb.com/rooms/1686958  20160104002432   
1014   9416373   https://www.airbnb.com/rooms/9416373  20160104002432   
847    7648716   https://www.airbnb.com/rooms/7648716  20160104002432   
1234   1472532   https://www.airbnb.com/rooms/1472532  20160104002432   
2882    430453    https://www.airbnb.com/rooms/430453  20160104002432   
236    1799255   https://www.airbnb.com/rooms/1799255  20160104002432   

     last_scraped                                 name  \
1528   2016-01-04      Historic loft in Pioneer Square   
2012   2016-01-04  Green Lake/Phinney/Greenwood Nugget   
1067   2016-01-04  Best Ocean Views By Pike Place! PA1   
2556   2016-01-04     Private Suite w Jacuzzi near UW!   
2767   2016-01-04         Gorgeous Old Style Apartment   
1014   2016-01-04         One bedroom… Great location.   
847    2016-01-04  Central Dist King Bd & Ensuite Bath   
1234   2016-01-04  Downtown-5 mins walk to Pike Place!   
2882   2016-01-04     Space Needle Views, Downtown Apt   
236    2016-01-04              Cozy Queen Anne Cottage   

                                                summary  \
1528  Located in the heart of historic Pioneer Squar...   
2012  Our freshly updated basement apartment will be...   
1067  Post Alley Court offers amazing views of Ellio...   
2556  Spacious bedroom and den with separate entry i...   
2767                                                NaN   
1014  One bedroom apartment close to downtown Seattl...   
847   In Aug and Sept this 2nd floor master bedroom ...   
1234                                                NaN   
2882                                                NaN   
236   Stay in one of the best neighborhoods in Seatt...   

                                                  space  \
1528                                                NaN   
2012  The perfect cozy spot for adventures all over ...   
1067  Water views from balcony, only 1/2 block to Pi...   
2556  We are world travelers and long-time Seattle r...   
2767  Our apartment is part of a large craftsman hom...   
1014                                                NaN   
847   Our home is simple and comfortable.  Our rooms...   
1234  We at Emerald Stays welcome you to our modern ...   
2882  A classic, 700+sq ft, 1bed / 1 bath apartment ...   
236   My house was built in 1904 but has been remode...   

                                            description experiences_offered  \
1528  Located in the heart of historic Pioneer Squar...                none   
2012  Our freshly updated basement apartment will be...                none   
1067  Post Alley Court offers amazing views of Ellio...                none   
2556  Spacious bedroom and den with separate entry i...                none   
2767  Our apartment is part of a large craftsman hom...                none   
1014  One bedroom apartment close to downtown Seattl...                none   
847   In Aug and Sept this 2nd floor master bedroom ...                none   
1234  We at Emerald Stays welcome you to our modern ...                none   
2882  A classic, 700+sq ft, 1bed / 1 bath apartment ...                none   
236   Stay in one of the best neighborhoods in Seatt...                none   

                                  neighborhood_overview  ...  \
1528  I love the old brick buildings and romance of ...  ...   
2012  This neighborhood is awesome. A vibrant area w...  ...   
1067  In the heart of Downtown Seattle! Nearby Attra...  ...   
2556  We are located on a quiet residential street i...  ...   
2767                                                NaN  ...   
1014           

##### Overview of listings_df

It has 91 columns

In [11]:
# Get an Overview of columns that are float
listings_df.loc[:,(listings_df.dtypes=='float64').values]

host_listings_count  host_total_listings_count   latitude   longitude  \
0                     3.0                        3.0  47.636289 -122.371025   
1                     6.0                        6.0  47.639123 -122.365666   
2                     2.0                        2.0  47.629724 -122.369483   
3                     1.0                        1.0  47.638473 -122.369279   
4                     2.0                        2.0  47.632918 -122.372471   
...                   ...                        ...        ...         ...   
3813                354.0                      354.0  47.664295 -122.359170   
3814                  1.0                        1.0  47.649552 -122.318309   
3815                  1.0                        1.0  47.508453 -122.240607   
3816                  1.0                        1.0  47.632335 -122.275530   
3817                  1.0                        1.0  47.641186 -122.342085   

      bathrooms  bedrooms  beds  square_feet  review_scores_rating  \
0           1.0       1.0   1.0          NaN                  95.0   
1           1.0       1.0   1.0          NaN                  96.0   
2           4.5       5.0   7.0          NaN                  97.0   
3           1.0       0.0   2.0          NaN                   NaN   
4           2.0       3.0   3.0          NaN                  92.0   
...         ...       ...   ...          ...                   ...   
3813        2.0       3.0   3.0          NaN                  80.0   
3814        1.0       1.0   2.0          NaN                 100.0   
3815        1.0       1.0   1.0          NaN                   NaN   
3816        1.0       0.0   1.0          NaN                   NaN   
3817        1.5       2.0   1.0          NaN                   NaN   

      review_scores_accuracy  review_scores_cleanliness  \
0                       10.0                       10.0   
1                       10.0                       10.0   
2                       10.0                       10.0   
3                        NaN                        NaN   
4                        9.0                        9.0   
...                      ...                        ...   
3813                     8.0                       10.0   
3814                    10.0                       10.0   
3815                     NaN                        NaN   
3816                     NaN                        NaN   
3817                     NaN                        NaN   

      review_scores_checkin  review_scores_communication  \
0                      10.0                         10.0   
1                      10.0                         10.0   
2                      10.0                         10.0   
3                       NaN                          NaN   
4                      10.0                         10.0   
...                     ...                          ...   
3813                    4.0                          8.0   
3814                   10.0                         10.0   
3815                    NaN                          NaN   
3816                    NaN                          NaN   
3817                    NaN                          NaN   

      review_scores_location  review_scores_value  license  reviews_per_month  
0                        9.0                 10.0      NaN               4.07  
1                       10.0                 10.0      NaN               1.48  
2                       10.0                 10.0      NaN               1.15  
3                        NaN                  NaN      NaN                NaN  
4                        9.0                  9.0      NaN               0.89  
...                      ...                  ...      ...                ...  
3813                    10.0                  8.0      NaN               0.30  
3814                    10.0                 10.0      NaN               2.00  
3815                     NaN                  NaN      NaN        

In [14]:
test=(listings_df.dtypes=='object').values
type(test)

numpy.ndarray

In [12]:
# Get an Overview of columns that are Object
listings_df.loc[:,(listings_df.dtypes=='object').values]

listing_url last_scraped  \
0       https://www.airbnb.com/rooms/241032   2016-01-04   
1       https://www.airbnb.com/rooms/953595   2016-01-04   
2      https://www.airbnb.com/rooms/3308979   2016-01-04   
3      https://www.airbnb.com/rooms/7421966   2016-01-04   
4       https://www.airbnb.com/rooms/278830   2016-01-04   
...                                     ...          ...   
3813   https://www.airbnb.com/rooms/8101950   2016-01-04   
3814   https://www.airbnb.com/rooms/8902327   2016-01-04   
3815  https://www.airbnb.com/rooms/10267360   2016-01-04   
3816   https://www.airbnb.com/rooms/9604740   2016-01-04   
3817  https://www.airbnb.com/rooms/10208623   2016-01-04   

                                     name  \
0            Stylish Queen Anne Apartment   
1      Bright & Airy Queen Anne Apartment   
2     New Modern House-Amazing water view   
3                      Queen Anne Chateau   
4          Charming craftsman 3 bdm house   
...                                   ...   
3813   3BR Mountain View House in Seattle   
3814    Portage Bay View!-One Bedroom Apt   
3815    Private apartment view of Lake WA   
3816    Amazing View with Modern Comfort!   
3817            Large Lakefront Apartment   

                                                summary  \
0                                                   NaN   
1     Chemically sensitive? We've removed the irrita...   
2     New modern house built in 2013.  Spectacular s...   
3     A charming apartment that sits atop Queen Anne...   
4     Cozy family craftman house in beautiful neighb...   
...                                                 ...   
3813  Our 3BR/2BA house boasts incredible views of t...   
3814  800 square foot 1 bedroom basement apartment w...   
3815  Very comfortable lower unit. Quiet, charming m...   
3816  Cozy studio condo in the heart on Madison Park...   
3817  All hardwood floors, fireplace, 65" TV with Xb...   

                                                  space  \
0     Make your self at home in this charming one-be...   
1     Beautiful, hypoallergenic apartment in an extr...   
2     Our house is modern, light and fresh with a wa...   
3                                                   NaN   
4     Cozy family craftman house in beautiful neighb...   
...                                                 ...   
3813  Our 3BR/2BA house bright, stylish, and wheelch...   
3814  This space has a great view of Portage Bay wit...   
3815                                                NaN   
3816  Fully furnished unit to accommodate most needs...   
3817                                                NaN   

                                            description experiences_offered  \
0     Make your self at home in this charming one-be...                none   
1     Chemically sensitive? We've removed the irrita...                none   
2     New modern house built in 2013.  Spectacular s...                none   
3     A charming apartment that sits atop Queen Anne...                none   
4     Cozy family craftman house in beautiful neighb...                none   
...                                                 ...                 ...   
3813  Our 3BR/2BA house boasts incredible views of t...                none   
3814  800 square foot 1 bedroom basement apartment w...                none   
3815  Very comfortable lower unit. Quiet, charming m...                none   
3816  Cozy studio condo in the heart on Madison Park...                none   
3817  All hardwood floors, fireplace, 65" TV with Xb...                none   

                                  neighborhood_overview  \
0                                                   NaN   
1     Queen Anne is a wonderful, truly functional vi...   
2     Upper Queen Anne is a charming neighborhood fu...   
3                                                   NaN   
4     We are in the beautiful neighborhood of Queen ...   
...                                              

In [ ]:
listings_df.loc[:,(listings_df.dtypes=='int').values]

In [ ]:
listings_df.loc[:,(listings_df.dtypes=='datetime').values]

There are no datetime columns in the listings_df. Clever to have separated the datetime. 

In [ ]:
listings_df.columns

Due to the Number of columns, working with this table might be cumbersome. I will normalize the listings_df table into smaller tables with id being the key to the relationship

##### Overview of calendar_df

It has four columns

In [ ]:
calendar_df.loc[:,(calendar_df.dtypes=='datetime').values]

In [ ]:
print(calendar_df.columns)
calendar_df.info()

Column 'date' is not in right datatype

In [ ]:
calendar_df.date.dtypes

##### Overview of review_df

It has six columns

In [ ]:
reviews_df.info()

Column 'date' is not in right datatype

#### Data Cleaning and Normalization

Convert date to appropriate datatype

In [ ]:
# Convert calendar_df Date
calendar_df['date'] = pd.to_datetime(calendar_df.date, yearfirst=True)
calendar_df['date'].dtypes

In [ ]:
reviews_df['date'] = pd.to_datetime(reviews_df.date, yearfirst=True)
reviews_df['date'].dtypes

##### Check for duplicates

In [8]:
print(reviews_df.duplicated().sum())
print(calendar_df.duplicated().sum())
print(listings_df.duplicated().sum())

0
0
0


##### Data Normalization of Listing DataFrame

In [ ]:
# Normalize for listings_location for locational details

listings_location = listings_df[['id',
'name', 
'description', 
'neighbourhood', 
'city', 
'state', 
'is_location_exact',
 'zipcode',
 'market',
 'smart_location',
 'country_code',
 'country',
 'latitude',
 'longitude']]

listings_location.shape

listings_location has 9 columns 

In [ ]:
# Normalize for listings_price for pricing details
listings_price = listings_df[['id', 
'price',
'weekly_price',
'monthly_price',
'security_deposit',
'cleaning_fee']]

listings_price.shape

In [ ]:
listings_df[['guests_included','extra_people']]


In [ ]:
listings_df[['minimum_nights',                    
'maximum_nights',                       
'has_availability',                
'availability_30',                
'availability_60',                    
'availability_90',   
'availability_365']]

listings_price has 8 columns

In [ ]:
# Normalize for listings_bedtails for bedding details

listings_bedtails = listings_df[['id', 
'property_type',
'room_type',
'accommodates',
'bathrooms',
'bedrooms',
'beds',
'bed_type',
'amenities',
'square_feet',
'amenities',
'square_feet']]

listings_bedtails

listings_bedtails has 8 columns 

In [ ]:
listings_hosts = listings_df[['id', 'host_id',
 'host_url',
 'host_name',
 'host_since',
 'host_location',
 'host_about',
 'host_response_time',
 'host_response_rate',
 'host_acceptance_rate',
 'host_is_superhost',
 'host_thumbnail_url',
 'host_picture_url',
 'host_neighbourhood',
 'host_listings_count',
 'host_total_listings_count',
 'host_verifications',
 'host_has_profile_pic',
 'host_identity_verified']]

listings_hosts.shape

listings_hosts has 19 columns

In [ ]:
listings_review = listings_df[['id', 'number_of_reviews',
 'first_review',
 'last_review',
 'review_scores_rating',
 'review_scores_accuracy',
 'review_scores_cleanliness',
 'review_scores_checkin',
 'review_scores_communication',
 'review_scores_location',
 'review_scores_value']]


# listings_review
listings_review.shape

In [ ]:
listings_review.sample(10)

listings_review has 11 columns

In [ ]:
11+19+8+8+16+23

In [ ]:
type(list(listings_bedtails.columns))

In [ ]:

# for x in list2:
#   list1.append(x)



# for x in list2:
#   list1.append(x)

init_listings_col = list(listings_bedtails.columns).extend(listings_hosts.columns.values)
# type(init_listings_col)


# print(init_listings_col)
# for i in :
#     for i in listings_df.columns.to_list:
#     if 

In [ ]:
listings_others=listings_df[['guests_included',
'extra_people',
'minimum_nights',
'maximum_nights',
'calendar_updated',
'has_availability',
'availability_30',
'availability_60',
'availability_90',
'availability_365',
'calendar_last_scraped','requires_license',
'license',
'jurisdiction_names',
'instant_bookable',
'cancellation_policy',
'require_guest_profile_picture',
'require_guest_phone_verification',
'calculated_host_listings_count',
'reviews_per_month']]

listings_others.shape

### Exploratory Data Analysis

First, I will calculate the occupancy rate of each listings. Occupancy Rate is a very important metric for short stays as it shows how well your apartment is performing, a High Occupancy Rate is considered good.
* To get the Occupancy Rate for each listing, I divide the Number of days the listing was occupied by the total period
* For total period, It is worthy to note that it is possible that listings might not have the same start date. To account for this, I will get the date difference for each listing; the difference between date of first instance and date of last instance

In [ ]:
# add aggregation by first and last 
df_agg= calendar_df.groupby(['listing_id']).agg({'date':['first','last']})

# Flatten Multiindex
df_agg.columns = df_agg.columns.map('_'.join)


# Convert to Date time
df_agg[['date_first','date_last']] = df_agg[['date_first','date_last']].apply(pd.to_datetime) #if conversion required

# Find Date difference
df_agg['date_diff'] = (df_agg['date_last'] - df_agg['date_first'])
df_agg.sample(10)

In [ ]:
df_agg['date_last'].value_counts()

* All the listings have the same date difference. Now, I can finally proceed to calculating the occupancy rate using 365 days as the denominator

In [ ]:
calendar_df

In [ ]:
# Calculate the Occupancy Count i.e. (where 'available' is 't')
occupancy = calendar_df.groupby('listing_id')['available'].apply(lambda x: (x=='t').sum()).reset_index(name='count')

# Create 'occupancy_rate' feature for each Listing.
occupancy['rate'] = (occupancy['count']*100)/365
occupancy.sample(20)

In [ ]:
# Get Distribution of occupancy rate
# plt.figure(figsize=(12,7))
plt.rcParams['figure.figsize']=[12,7]
occupancy['rate'].plot(kind='hist', bins=20)
plt.title('Distribution of Occupancy Rate')
plt.show()

Apparently, there are listings with a perfect record i.e. 100% occupancy rate and this group represents the majority 

I will get a table for all listings with a perfect record.

In [ ]:
# Get a table of all listings with a perfect record

perfect_rec = occupancy[occupancy['rate'] == 100]
perfect_rec.info()

There are 678 listings with a perfect record 

I feel pressured to take a deeper dive to understand the characteristics of this particular group, but before then, I will peruse through the normalized llisting DataFrames

In [ ]:
# Check bed details 

listings_bedtails.sample(10)

In [ ]:

# Plot for Property Type
# Choose color for plot
def count_univ(data, y, color= sns.color_palette()[0]):
    order = data[y].value_counts().index

    sns.countplot(data=data, y=y, color=color, order=order)
    plt.title(f'Frequency of {y}')
    plt.ylabel(f'{y}', fontsize=12)
    plt.xlabel('Count', fontsize=12)
    plt.show()
count_univ(data=listings_bedtails,y='property_type')

In general, There are more houses than any other property type for short stay accommodation in Seattle. Following closely are Apartments.

The Margin for error is not known as it's common for owners when uploading to use a generic term 'House' for the property description



In [ ]:
# Plot for Room Type
count_univ(data=listings_bedtails,y='room_type')

As expected there are more listings let out in their entirety that there are shared or private rooms

Next is to check for bedrooms

In [ ]:
# Plot for Room Type
count_univ(data=listings_bedtails,y='bedrooms')

First thing to note from the visualization is the 0 bedrooms. This could be studio apartment that have both the bedroom and the living area as one single area.

Also, 1-Bedrooms are the most frequent bedroom type, representing majority of the dataset 

Now, merging the listing_bedtails dataframe with my perfect_rec dataframe, I can get the bed details for the perfect record listings and see how they are represented in my perfect_rec dataframe

In [ ]:
# Merge Bed Details with Perfect record
temp_listings = perfect_rec.merge(listings_bedtails, how='left',left_on='listing_id',right_on='id',suffixes=('', '_drop'))

#Drop the duplicate columns
temp_listings.drop([col for col in temp_listings.columns if 'drop' in col], axis=1, inplace=True)




Plot for bedrooms for listings with Perfect Records

In [ ]:
# base_color = sns.color_palette()[0]

# temp_listings.bedrooms.value_counts(normalize=True).plot(kind='bar', color=base_color)
# # plt.xticks(rotation=45)
# temp_listings

# Plot for bedrooms for listings with Perfect Records
count_univ(data=temp_listings,y='bedrooms')

Apart from the fact that there are no 6 or 7 Bedrooms with a Perfect Record. This chart is similar to the one for all listings

In [ ]:
# Plot for Room Type for listings with Perfect Records
count_univ(data=temp_listings,y='room_type')

This is interesting, compared to the chart for all listings, private rooms have a very good representation when it comes to having Perfect Records. 

I would love to see the ratio of Private Rooms with Perfect Records to All Private Rooms

In [ ]:
# Plot for Property Type for listings with Perfect Records
count_univ(data=temp_listings,y='property_type')

The gap between the two leading property type has increased.

However, there are still more **Houses** with Perfect Records than any other property type for short stay accommodation in Seattle. Following closely are Apartments.

The Margin for error is not known as it's common for owners when uploading to use a generic term 'House' for the property description



In [ ]:
calendar_df

In [ ]:
# df_agg['diff'].value_counts()

In [ ]:
# temp_listings = listings_max.merge(listings_bedtails, how='left',left_on='id',right_on='id',suffixes=('', '_drop'))

# #Drop the duplicate columns
# temp_listings.drop([col for col in temp_listings.columns if 'drop' in col], axis=1, inplace=True)

# base_color = sns.color_palette()[0]

# temp_listings.beds.value_counts(normalize=True).plot(kind='bar', color=base_color)
# # plt.xticks(rotation=45)
# # temp_listings


In [ ]:
reviews_df.sample(10)

In [ ]:
reviews_count = reviews_df.groupby('listing_id')['listing_id'].count().sort_values(ascending=False).reset_index(name='review_count')

In [ ]:
reviews_count

In [ ]:
# occupancy_review = occupancy.merge(reviews_count, how='left',left_on='listing_id',right_on='listing_id',suffixes=('', '_drop'))


# x= occupancy_review['review_count']
# y = occupancy_review['rate']

# bins = np.arange(x.min(),x.max(),0.5)
# sns.regplot(data=occupancy_review, x=x, y=y,scatter_kws={'alpha': 1/20}, bins=bins)